File: test/test_discard.cpp
#include <stdio.h>
#include "TRTCContext.h"
#include "fake_vectors/DVCounter.h"
#include "fake_vectors/DVDiscard.h"
#include "transform.h"

int main()
{	
	TRTC_Set_Verbose();

	// just to verify that it compiles
	DVDiscard sink("int32_t");
	TRTC_Transform(DVCounter(DVInt32(5), 10), sink, Functor("Negate"));

	return 0;
}

File: CSharp/test/test_discard/test_discard.cs
using ThrustRTCSharp;

namespace test_discard
{
    class test_discard
    {
        static void Main(string[] args)
        {
            TRTC.Set_Verbose();

            // just to verify that it compiles
            DVCounter src = new DVCounter(new DVInt32(5), 10);
            DVDiscard sink = new DVDiscard("int32_t");
            TRTC.Copy(src, sink);

        }
    }
}
